packages/react/use-funnel/src/useFunnel.tsx

```diff
@@ -56,13 +56,16 @@ export const useFunnel = <Steps extends NonEmptyArray<string>>(
     () =>
       Object.assign(
         function RouteFunnel(props: RouteFunnelProps<Steps>) {
-          // eslint-disable-next-line react-hooks/rules-of-hooks
-          const step = useQueryParam<Steps[number]>(stepQueryKey) ?? options?.initialStep;
+          const step = useQueryParam(stepQueryKey) ?? options?.initialStep;
 
           assert(
             step != null,
             `표시할 스텝을 ${stepQueryKey} 쿼리 파라미터에 지정해주세요. 쿼리 파라미터가 없을 때 초기 스텝을 렌더하려면 useFunnel의 두 번째 파라미터 options에 initialStep을 지정해주세요.`
           );
+          assert(
+            !Array.isArray(step),
+            `표시할 스텝을 ${stepQueryKey} 쿼리 파라미터에 배열타입으로 지정하면 안됩니다. string타입으로 지정해주세요.`
+          );
```

Comment:
step can be string[] type, so I did type narrowing with removing edgecase of queryparam